src/components/layout/PrimaryLayout/PrimaryLayout.js

```diff
@@ -9,85 +9,106 @@ import Header from 'components/Header'
 import AlphaFlag from 'components/AlphaFlag'
 import HposSettingsQuery from 'graphql/HposSettingsQuery.gql'
 import useConnectionContext from 'contexts/useConnectionContext'
+import useHFConnectionContext from 'holofuel/contexts/useConnectionContext'
 import useFlashMessageContext from 'contexts/useFlashMessageContext'
 import useCurrentUserContext from 'contexts/useCurrentUserContext'
-import { useInterval } from 'utils'
+import { useInterval, POLLING_INTERVAL_SETTINGS } from 'utils'
 import { wsConnection } from 'holochainClient'
 import { isLoginPage, HP_ADMIN_LOGIN_PATH } from 'utils/urls'
 import styles from './PrimaryLayout.module.css' // eslint-disable-line no-unused-vars
 import 'global-styles/colors.css'
 import 'global-styles/index.css'
 
-export function PrimaryLayout ({
+function PrimaryLayout ({
   children,
   headerProps = {},
   showHeader = true,
   showAlphaFlag = true
 }) {
   const [isInsideApp, setIsInsideApp] = useState(true)
   const [isHposConnectionAlive, setIsHposConnectionAlive] = useState(true)
-  const { setConnectionStatus, connectionStatus } = useConnectionContext()
+  const { connectionStatus, setConnectionStatus } = useConnectionContext()
+  const { isConnected: isHFConductorConnected } = useHFConnectionContext()
   const { setCurrentUser } = useCurrentUserContext()
   const { newMessage } = useFlashMessageContext()
   const { push } = useHistory()
 
+  const [isPausedConnectionCheckInterval, setIsPausedConnectionCheckInterval] = useState(false)
+  const [userMessage, setUserMessage] = useState('')
```

Comment:
I'm not seeing where the user is ever seeing the value of this state, so calling it `userMessage` makes the code harder to read and maintain. AFAICT, what this state is doing is just tracking which of 3 states the app is in, so I might just call it `connectionStatus`, and use constants to populate it.

```
const [connectionStatus, setConnectionStatus] = useState(CONNECTION_STATUS.connected)
```

and then eg on line 75

```
setConnectionStatus(CONNECTION_STATUS.noHoloport)
```

I think this makes it much clearer what's going on.

Reply:
On line 76, 87, and 93, the `userMessage` is set to `newMessage` to render a flash message with the current user message.

Reply:
right, but there's no reason to keep that in state.

You're not going

`setNewMessage(userMessage)`

you're going

```
setUserMessage(someConstant)
setNewMessage(someConstant)
```

The fact that you're storing the same constant in `userMessage` as you happen to be displaying to the user is irrelevant and confusing.

Reply:
Got it... ok, I can see how that might be clearer.  Updates soon to come. :+1: